File: c2_datatype/src/main/java/test/followme/Testnull.java
package test.followme;

import io.reactivex.rxjava3.annotations.NonNull;

import java.math.BigDecimal;

import static test.followme.DoubleUtil.formatDecimal;

public class Testnull {
    public static void main(String[] args) {
        String str=null;
        boolean isString=null instanceof String;
        System.out.println(isString);
        System.out.println("=================================");
    }
}

File: c2_datatype/src/main/java/test/followme/TestNull.kt
package test.followme

fun main() {

//    println(tString())

    var s1=null

    println(s1 as String?)
    println(s1 as? String ?:111)
    println(s1 as? String ?:111)
}

fun tString():String{
    //return null
    return DoubleUtil.name
}